### ISSUE TITLE
pallet-mmr: Discuss reducing the number of methods in runtime API and RPC API

### ISSUE THREAD

**Issue by serban300** [paritytech/substrate#12391]
Since we're changing the pallet-mmr runtime & RPC APIs as part of #12339 it's a good opportunity to revisit some other decisions related to the API:

1. Whether we should have 2 separate methods `generate_batch_proof()` and `generate_historical_batch_proof()`. Since in the end `generate_batch_proof()` is a special case of `generate_historical_batch_proof()`. See the discussion here: https://github.com/paritytech/substrate/pull/12324#discussion_r976692621
2. If we should keep generate_proof(), verify_proof(), and verify_proof_stateless() since all these can be done via batching methods (e.g. generate_batch_proof(), etc)

cc: @acatangiu @Lederstrumpf  @svyatonik

**Comment by acatangiu** [comment]
I agree with this, but drop "batching" or "historical" from new "universal" API name, should be simple generate_proof(), verify_proof(), and verify_proof_stateless() with optional extra params.
Documentation should also be very clear to explain specific param combinations.

**Comment by Lederstrumpf** [comment]
> I agree with unifying all, but drop "batching" or "historical" from new "universal" API name, should be simple generate_proof(), verify_proof(), and verify_proof_stateless() with optional extra params.
Documentation should also be very clear to explain specific param combinations.

ditto: I support unification, and if unified, then batching & historical qualifiers are superfluous and should be dropped.

**Comment by serban300** [comment]
Sounds good ! I also agree with these naming scheme.
